### DISCUSSION TITLE
Proxmox Automated Install

### DISCUSSION THREAD

**Discussion by irish1986** [tteck/Proxmox#3289] [Ideas]
So after a few weeks working on Proxmox Automated Installation (pve bare metal provisioning) I was finally successful at it.

The official docs explains everything but can be slightly confusing at time. I wonder how this could be add to this excellent collection of script to make it easier.

There are a few way to achieve it, I was able to get it working serving a toml file over the network filtered on MAC adress.

**Comment by alexanderjerome** [comment]
I would absolutely love your pointers on setting up automated install, irish. I had some difficulties with this as well and have sorely needed it. I have a remote machine in another country, but the motherboard's USB ports being faulty it's not possible to use a keyboard and the USB installation drive simultaneously! Do you mind sharing the instructions you followed and provide some tips? Thanks!

Seems to me that when it comes to the helper scripts, I could see the utility in a bash script that could capture certain elements of a PVE hosts configuration, and then package that into an ISO. Then, you could transfer that into the ISO directory of the iVentoy container. It would be a great utility to have, a script that can help a user package an automated installer. +1 on this feature

**Reply by irish1986** [reply]
I will work on documenting and getting a basic framework for it. Proxmox supports a few ways to get a automated baremetal system installation running but only a few makes sense to me from a homelab point of view

I got curious to get this working especially given I nuked my homelab composed of ~12 nodes (all proxmox) a few weeks ago in some catastrophic events. Long story settintup twelve pve environmental is time-consuming.

I the end I would envision a helper script to kick start the baremetal installation and if possible trigger some other helper script. That second part needs research on my end though.

**Reply by alexanderjerome** [reply]
That's horrible. I can only imagine the work you'll have to do. Although PBS could have been a good solution for you, it's showing that having a remote server with enough storage space is essential in those cases.

I've seen some tutorials on YT that use Ansible for that. I've also wondered why the Proxmox scripts don't use Ansible, but I can imagine it's because of the effort to keep external dependencies at a minimum on the host machine?

I'm eager to see this happen and would be happy to contribute / test such a feature in collaboration. I often need to install servers in remote locations and the people I'm generally working with aren't Linux savvy or know what they're doing most of the time.

My understanding is that the official way to do this is to set up a automated debian install and then have a script install the required libraries, but a direct standalone version would be ideal. I can see the flow starting with a specific version of Proxmox (8.1 and above), and then running an pre-configured version of the Post-Install helper script.

I mentioned this in #3280 , being able to save presets for how we'd like our containers to be configured can work will with an automated install script. This could facilitate not only automating the PVE host installation, but in my case of remote installation, be able to auto install Tailscale+Headscale, connect to PBS and install container templates, or even add a remote NFS server storage option for accessing images right out of the box, configuring OpenID authentication to Proxmox, etc. So many configurations are tedious but often require exact repetition.
